### ISSUE TITLE
task generated an expection

### ISSUE THREAD

**Issue by Moranyixiao** [mcpvp-club/FasterCrystals#9]
![8 XOSO4(XM%{E F0~$1~4L5](https://github.com/mcpvp-club/FasterCrystals/assets/26624650/0cc138ab-b534-4032-9059-d13b911431bc)

**Comment by Moranyixiao** [comment]
Using Purpur 2141

```
[15:05:05] [Server thread/WARN]: [FasterCrystals] Task #16552266 for FasterCrystals v1.4.2 generated an exception
java.lang.IllegalStateException: Cannot convert 'DamageType[msgId=player, scaling=WHEN_CAUSED_BY_LIVING_NON_PLAYER, exhaustion=0.1, effects=HURT, deathMessageType=DEFAULT]' to bukkit representation, since it is not registered.
	at org.bukkit.craftbukkit.v1_20_R3.CraftRegistry.lambda$minecraftToBukkit$0(CraftRegistry.java:67) ~[purpur-1.20.4.jar:git-Purpur-2141]
	at java.util.Optional.orElseThrow(Optional.java:403) ~[?:?]
	at org.bukkit.craftbukkit.v1_20_R3.CraftRegistry.minecraftToBukkit(CraftRegistry.java:67) ~[purpur-1.20.4.jar:git-Purpur-2141]
	at org.bukkit.craftbukkit.v1_20_R3.damage.CraftDamageType.minecraftToBukkit(CraftDamageType.java:123) ~[purpur-1.20.4.jar:git-Purpur-2141]
	at org.bukkit.craftbukkit.v1_20_R3.damage.CraftDamageType.minecraftHolderToBukkit(CraftDamageType.java:102) ~[purpur-1.20.4.jar:git-Purpur-2141]
	at org.bukkit.craftbukkit.v1_20_R3.damage.CraftDamageSource.<init>(CraftDamageSource.java:21) ~[purpur-1.20.4.jar:git-Purpur-2141]
	at org.bukkit.craftbukkit.v1_20_R3.event.CraftEventFactory.handleEntityDamageEvent(CraftEventFactory.java:1092) ~[purpur-1.20.4.jar:git-Purpur-2141]
	at org.bukkit.craftbukkit.v1_20_R3.event.CraftEventFactory.callNonLivingEntityDamageEvent(CraftEventFactory.java:1266) ~[purpur-1.20.4.jar:git-Purpur-2141]
	at org.bukkit.craftbukkit.v1_20_R3.event.CraftEventFactory.handleNonLivingEntityDamageEvent(CraftEventFactory.java:1270) ~[purpur-1.20.4.jar:git-Purpur-2141]
	at org.bukkit.craftbukkit.v1_20_R3.event.CraftEventFactory.handleNonLivingEntityDamageEvent(CraftEventFactory.java:1256) ~[purpur-1.20.4.jar:git-Purpur-2141]
	at net.minecraft.world.entity.boss.enderdragon.EndCrystal.hurt(EndCrystal.java:144) ~[?:?]
	at xyz.reknown.fastercrystals.damager.CrystalDamager_1_20_R3.damage(CrystalDamager_1_20_R3.java:15) ~[pvpclub-FasterCrystals-1.4.2.jar:?]
	at xyz.reknown.fastercrystals.listeners.packet.AnimationListener.lambda$onPacketPlayReceive$1(AnimationListener.java:86) ~[pvpclub-FasterCrystals-1.4.2.jar:?]
	at org.bukkit.craftbukkit.v1_20_R3.scheduler.CraftTask.run(CraftTask.java:101) ~[purpur-1.20.4.jar:git-Purpur-2141]
	at org.bukkit.craftbukkit.v1_20_R3.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:482) ~[purpur-1.20.4.jar:git-Purpur-2141]
	at net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1669) ~[purpur-1.20.4.jar:git-Purpur-2141]
	at net.minecraft.server.dedicated.DedicatedServer.tickChildren(DedicatedServer.java:487) ~[purpur-1.20.4.jar:git-Purpur-2141]
	at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1548) ~[purpur-1.20.4.jar:git-Purpur-2141]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1236) ~[purpur-1.20.4.jar:git-Purpur-2141]
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:323) ~[purpur-1.20.4.jar:git-Purpur-2141]
	at java.lang.Thread.run(Thread.java:1583) ~[?:?]
```

**Comment by Jyguy** [comment]
Fixed in 1.4.3, thanks for the bug report!
